@xen-orchestra/web-core/lib/components/ui/table-pagination/UiTablePagination.vue

```diff
@@ -0,0 +1,207 @@
+<!-- v2 -->
+<template>
+  <div class="ui-table-pagination">
+    <span class="typo p3-regular label">
+      {{ $t('core.select.n-object-of', { from: startIndex, to: endIndex, total: totalItems }) }}
+    </span>
+
+    <div class="buttons-content">
+      <UiButtonIcon
+        :disabled="isFirstPage"
+        accent="info"
+        class="icon"
+        size="small"
+        :icon="faAngleDoubleLeft"
+        @click="goToFirstPage()"
+      />
+      <UiButtonIcon
+        :disabled="isFirstPage"
+        accent="info"
+        class="icon"
+        size="small"
+        :icon="faAngleLeft"
+        @click="prev()"
+      />
+      <UiButtonIcon
+        :disabled="isLastPage"
+        accent="info"
+        class="icon"
+        size="small"
+        :icon="faAngleRight"
+        @click="next()"
+      />
+      <UiButtonIcon
+        :disabled="isLastPage"
+        accent="info"
+        class="icon"
+        size="small"
+        :icon="faAngleDoubleRight"
+        @click="goToLastPage()"
+      />
+    </div>
+
+    <span class="typo p3-regular label">{{ $t('core.show-by') }}</span>
+    <div class="dropdown-wrapper">
+      <select v-model="pageSize" class="dropdown typo c3-regular" @change="updatePagination">
+        <option v-for="option in pageSizeOptions" :key="option" :value="option" class="typo p2-medium">
+          {{ option }}
+        </option>
+      </select>
+      <VtsIcon class="icon" accent="current" :icon="faAngleDown" />
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import VtsIcon from '@core/components/icon/VtsIcon.vue'
+import UiButtonIcon from '@core/components/ui/button-icon/UiButtonIcon.vue'
+import {
+  faAngleDoubleLeft,
+  faAngleDoubleRight,
+  faAngleDown,
+  faAngleLeft,
+  faAngleRight,
+} from '@fortawesome/free-solid-svg-icons'
+import { useOffsetPagination } from '@vueuse/core'
+import { computed, ref } from 'vue'
+
+const { totalItems } = defineProps<{
+  totalItems: number
+  disabled?: boolean
+}>()
+
+const pageSize = ref(50)
+const pageSizeOptions = [10, 50, 100, 150, 200]
+const { currentPage, currentPageSize, pageCount, isFirstPage, isLastPage, prev, next } = useOffsetPagination({
```

Comment:
Let's try to keep consistency, following `goToFirstPage` and `goToLastPage` names:

```suggestion
const { currentPage, currentPageSize, pageCount, isFirstPage, isLastPage, prev: goToPreviousPage, next: goToNextPage } = useOffsetPagination({
```